### PR TITLE
[android] Allow target apk generation for Store use, bump SDK 30

### PR DESCRIPTION
**Pull request by fuzzard**
## Description
Introduce infrastructure capabilities to create and deploy an APK targeted at a specific deployment store.
Bump SDK 30

## Motivation and context
This allows us the ability to conditionally generate an apk that adheres to certain requirements for store deployment (ie Google Play).
A jenkins job will be created, that can be triggered to create a google play compatible apk (without MANAGE_EXTERNAL_STORAGE permission in this particular instance). This is generated by setting an environment variable at cmake generation time (eg ANDROIDSTORE=GOOGLE make -C tools/depends/target/cmakebuildsys) and an apk can then be created with a -GOOGLE set in the apk name to distinguish between an apk not built with the environment variable set.

Additionally, extract out min and target SDK numbers to a single cmake variable location. We could extend this to allow set at generation via a define, but for now, i think just setting them is suitable.

SDK 30 bump is done as per #21229 to use MANAGE_EXTERNAL_STORAGE to provide a "full featured" version when not deployed via Play store.

## How has this been tested?
Generation/build. manifest is correct in both instances, build naming is correct, and mirror deployment naming is now done (thanks @wsnipex )

## What is the effect on users?
The ability to deploy to the Google Playstore, and adhere to their "limitations". It also gives us an easy way to generate apk's (nightlies, releases, test builds, etc) that have the full kodi experience for local object manipulation.

Google play installs will be limited to only being able to read from local storage filetypes allowed by google. These limitations do NOT affect users who access their media/files from remote shares (ie nfs, smb, webdav, etc), only from things like physically attached USB storage to shields (or other devices), or "external storage" (SD cards, local locations) on phones.

## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [x] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [x] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

File: cmake/platform/android/android.cmake [hunk 1]
```diff
@@ -1,2 +1,14 @@
...
 set(APP_RENDER_SYSTEM gles)
+
+# Store SDK compile version
+set(TARGET_SDK 30)
+# Minimum supported SDK version
+set(TARGET_MINSDK 21)
+
+if(DEFINED ENV{ANDROIDSTORE} AND ("$ENV{ANDROIDSTORE}" STREQUAL "GOOGLE"))
+  set(PLAYSTORE_STATE "true")
+  set(PERMISSION.MANAGE_EXTERNAL_STORAGE "<uses-permission android:name=\"android.permission.MANAGE_EXTERNAL_STORAGE\" />")
```
Comment (ksooo) [inline cmake/platform/android/android.cmake:11]: I thought we do NOT want to set this permission for Playstore version?

**Comment by fuzzard** [comment]
Testbuild of the Playstore version: http://mirrors.kodi.tv/test-builds/android/arm64-v8a/kodi-20220520-a9ea38c2-android_storetype2-22879-google-arm64-v8a.apk

**Review by wsnipex — APPROVED** [review]

**Comment by joseluismarti** [comment]
It would be useful to add an info message in the log to easily identify which version of the apk is running

**Comment by garbear** [comment]
I think we should also show the "flavor" inside the app as discussed on Slack.

**Comment by fuzzard** [comment]
Yeah, was waiting on wording and placement confirmation on slack before i pushed it. save myself some time ;)

**Comment by Tolriq** [comment]
Not that anyone care about what I write, but you need to also handle version code or signature to avoid auto replace by different versions from the Stores.
If different signature then no issue (But users won't keep their data when migrating from Play Store to non Play Store version).
Else the usual way is to have a proper version code organisation with enough space for all the future version then use the higher bit to define the flavor, the lowest one going to Play Store to not replace the others.
Current version code = XXYYZZZ major/minor/ABI
Should then be XXXYYZZZ (so 1 more space for the major just in case) and final one being 1CODE for Play Store and 9CODE for no store, leaving space in the middle for possible other stores.

**Pull request closed without merging**